principles/code_is_not_yours.md

```diff
@@ -0,0 +1,74 @@
+# The code is not 'yours'
+
+> *We write and commit code thinking of someone else, not ourselves*
+
+When we refer to code we often do so in a personal manner; 'my code', 'my commit', 'the class I wrote'. There is no harm in this (and trying to avoid it might lead to funny conversations!) but the key principle to remember is
+
+> The code is not 'yours'
+
+When writing code you should be thinking of *the next developer* who has to work with it, who may be your future self, another member of the team, or someone completely new.
+
+They will not have the understanding and context you do at the time of writing it, so when they're amending or refactoring your work later it is important they can derive the intent of what your code is doing.
+
+There are things at our disposal to help with this.
+
+## Good naming
+
+Naming is hard.
+
+> There are only two hard things in Computer Science: cache invalidation and naming things.
+>
+> -- Phil Karlton
+
+That said you can avoid basic errors such as
+
+- Generic variable names: `p`, `temp`, `v`
+- Using arrays over hashes: `['Bob', 'Smith', 'Bristol']` over `{ name: 'Bob', surname: 'Smith', location: 'Bristol' }`
+- Use of acronyms: particularly domain specific ones. Someone new to the service may not know their GOR from their WIMS
+
+Other than that when considering the name for a method, class, module etc try not to approach it from what makes sense to you, but instead what best describes the intent of the thing, and what will help the person who comes after you.
+
+## Relevant documentation
+
+Lets be frank; few developers like writing documentation, whether its comments in the code or actual documentation like the README or accompanying wiki. So this principle doesn't advocate that **everything** should be commented. There is little value in documenting that `Person.name` gets and sets a person's name!
+
+### When writing code document the *why*
+
+Don't document what a thing is doing, that should be clear from the code. Document *why* the thing is needed, or *why* its doing it the way it is.
+
+### When documenting the project document the *how*
+
+When it comes to other developers or users of the project, the thing they are most interested in is the *how*. *How* can I build/deploy/run this project. *How* can I use it in my own work.
+
+## Clean commit history
+
+A clean commit history with meaningful messages is essential to understanding the intent of code and why its there.
```

Comment:
That's a fair point, 'intent' is the wrong choice in this context.

The only thing I'd add is that I believe we should be *aiming* to make the intent clear using clean, well designed, well named code.

Comments in my opinion should be reserved for when you have to get gnarly. So if the intent isn't clear, then add a comment in.

So wonder if its worth having an explicit section regards comments in code?